File: app/src/main/kotlin/com/dilarion/app/ui/components/EmojiCatalog.kt
package com.dilarion.app.ui.components

// Curated common-emoji set, grouped — not the full Unicode emoji database,
// but enough for a real picker (search + categories) without shipping a
// multi-thousand-entry dataset. Mirrors the desktop app's catalog.

data class EmojiCategory(val label: String, val emojis: List<String>)

val EMOJI_CATEGORIES: List<EmojiCategory> = listOf(
    EmojiCategory(
        "Smileys",
        listOf("😀", "😁", "😂", "🤣", "😊", "😇", "🙂", "🙃", "😉", "😍", "🥰", "😘", "😋", "😜", "🤪", "🤨", "🧐", "🤓", "😎", "🥳", "😏", "😒", "😞", "😔", "😢", "😭", "😤", "😡", "🤬", "😱", "😨", "😰", "😥", "🥺", "😴", "🤤", "😷", "🤒", "🤕", "🤢", "🥵", "🥶", "😵", "🤯", "🤠", "🥸", "😬", "🙄", "😑"),
    ),
    EmojiCategory(
        "Gestures",
        listOf("👍", "👎", "👌", "✌️", "🤞", "🤟", "🤘", "👊", "✊", "👏", "🙌", "👐", "🤲", "🙏", "💪", "🫡", "🖐️", "✋", "👋", "🤙", "☝️", "👆", "👇", "👈", "👉", "🫰", "🤝", "🖕"),
    ),
    EmojiCategory(
        "Hearts",
        listOf("❤️", "🧡", "💛", "💚", "💙", "💜", "🖤", "🤍", "🤎", "💔", "❣️", "💕", "💞", "💓", "💗", "💖", "💘", "💝"),
    ),
    EmojiCategory(
        "Animals",
        listOf("🐶", "🐱", "🐭", "🐹", "🐰", "🦊", "🐻", "🐼", "🐨", "🐯", "🦁", "🐮", "🐷", "🐸", "🐵", "🐔", "🐧", "🐦", "🦅", "🦉", "🐺", "🐗", "🐴", "🦄", "🐝", "🦋", "🐢", "🐍", "🦕", "🐙", "🐬", "🐳"),
    ),
    EmojiCategory(
        "Food",
        listOf("🍏", "🍎", "🍌", "🍉", "🍇", "🍓", "🍒", "🍑", "🥭", "🍍", "🥥", "🍅", "🍕", "🍔", "🍟", "🌭", "🥪", "🌮", "🌯", "🍣", "🍜", "🍝", "🍿", "🍩", "🍪", "🎂", "🍰", "🍫", "🍬", "🍭", "☕", "🍵", "🥤", "🍺", "🍷", "🥂"),
    ),
    EmojiCategory(
        "Activities",
        listOf("⚽", "🏀", "🏈", "⚾", "🎾", "🏐", "🎱", "🏓", "🏸", "🥊", "🎮", "🎲", "🎯", "🎳", "🎸", "🎤", "🎧", "🎨", "🎬", "📷", "✈️", "🚗", "🚀", "⛵"),
    ),
    EmojiCategory(
        "Objects",
        listOf("💡", "🔥", "✨", "🎉", "🎊", "🎁", "🏆", "⭐", "🌟", "💯", "✅", "❌", "⚠️", "🔒", "🔓", "📌", "📎", "💰", "💎", "⏰", "📱", "💻", "🔋", "🔑"),
    ),
)

fun searchEmoji(query: String): List<String> {
    val q = query.trim().lowercase()
    if (q.isEmpty()) return EMOJI_CATEGORIES.flatMap { it.emojis }
    val matched = EMOJI_CATEGORIES.filter { it.label.lowercase().contains(q) }
    return (if (matched.isNotEmpty()) matched else EMOJI_CATEGORIES).flatMap { it.emojis }
}

File: desktop/src/components/emojiCatalog.ts
// Curated common-emoji set, grouped — not the full Unicode emoji database,
// but enough for a real picker (search + categories) without shipping a
// multi-thousand-entry dataset.

export const EMOJI_CATEGORIES: { label: string; emojis: string[] }[] = [
  {
    label: 'Smileys',
    emojis: ['😀', '😁', '😂', '🤣', '😊', '😇', '🙂', '🙃', '😉', '😍', '🥰', '😘', '😋', '😜', '🤪', '🤨', '🧐', '🤓', '😎', '🥳', '😏', '😒', '😞', '😔', '😢', '😭', '😤', '😡', '🤬', '😱', '😨', '😰', '😥', '🥺', '😴', '🤤', '😷', '🤒', '🤕', '🤢', '🥵', '🥶', '😵', '🤯', '🤠', '🥸', '😬', '🙄', '😑'],
  },
  {
    label: 'Gestures',
    emojis: ['👍', '👎', '👌', '✌️', '🤞', '🤟', '🤘', '👊', '✊', '👏', '🙌', '👐', '🤲', '🙏', '💪', '🫡', '🖐️', '✋', '👋', '🤙', '☝️', '👆', '👇', '👈', '👉', '🫰', '🤝', '🖕'],
  },
  {
    label: 'Hearts',
    emojis: ['❤️', '🧡', '💛', '💚', '💙', '💜', '🖤', '🤍', '🤎', '💔', '❣️', '💕', '💞', '💓', '💗', '💖', '💘', '💝'],
  },
  {
    label: 'Animals',
    emojis: ['🐶', '🐱', '🐭', '🐹', '🐰', '🦊', '🐻', '🐼', '🐨', '🐯', '🦁', '🐮', '🐷', '🐸', '🐵', '🐔', '🐧', '🐦', '🦅', '🦉', '🐺', '🐗', '🐴', '🦄', '🐝', '🦋', '🐢', '🐍', '🦕', '🐙', '🐬', '🐳'],
  },
  {
    label: 'Food',
    emojis: ['🍏', '🍎', '🍌', '🍉', '🍇', '🍓', '🍒', '🍑', '🥭', '🍍', '🥥', '🍅', '🍕', '🍔', '🍟', '🌭', '🥪', '🌮', '🌯', '🍣', '🍜', '🍝', '🍿', '🍩', '🍪', '🎂', '🍰', '🍫', '🍬', '🍭', '☕', '🍵', '🥤', '🍺', '🍷', '🥂'],
  },
  {
    label: 'Activities',
    emojis: ['⚽', '🏀', '🏈', '⚾', '🎾', '🏐', '🎱', '🏓', '🏸', '🥊', '🎮', '🎲', '🎯', '🎳', '🎸', '🎤', '🎧', '🎨', '🎬', '📷', '✈️', '🚗', '🚀', '⛵'],
  },
  {
    label: 'Objects',
    emojis: ['💡', '🔥', '✨', '🎉', '🎊', '🎁', '🏆', '⭐', '🌟', '💯', '✅', '❌', '⚠️', '🔒', '🔓', '📌', '📎', '💰', '💎', '⏰', '📱', '💻', '🔋', '🔑'],
  },
];

export function searchEmoji(query: string): string[] {
  // No metadata to search by keyword with this small a catalog — search just
  // narrows to categories whose label matches, falling back to everything.
  const q = query.trim().toLowerCase();
  if (!q) return EMOJI_CATEGORIES.flatMap(c => c.emojis);
  const matched = EMOJI_CATEGORIES.filter(c => c.label.toLowerCase().includes(q));
  return matched.length ? matched.flatMap(c => c.emojis) : EMOJI_CATEGORIES.flatMap(c => c.emojis);
}
